crates/gpio/src/backend.rs

```diff
@@ -181,47 +212,21 @@ fn start_backend<D: 'static + GpioDevice + Send + Sync>(args: GpioArgs) -> Resul
             // threads, and so the code uses unwrap() instead. The panic on a thread won't cause
             // trouble to other threads/guests or the main() function and should be safe for the
             // daemon.
-            let controller = match cfg {
-                GpioDeviceType::PhysicalDevice(device_num) => match D::open(device_num) {
-                    Ok(device) => match GpioController::<D>::new(device) {
-                        Ok(controller) => controller,
-                        Err(error) => return Err(Error::CouldNotCreateGpioController(error)),
-                    },
-                    Err(error) => return Err(Error::CouldNotOpenDevice(error)),
-                },
-                _ => return Err(Error::DeviceUnimplemented),
-            };
-
-            let backend = Arc::new(RwLock::new(
-                VhostUserGpioBackend::new(controller).map_err(Error::CouldNotCreateBackend)?,
-            ));
-            let listener = Listener::new(socket.clone(), true).unwrap();
-
-            let mut daemon = VhostUserDaemon::new(
-                String::from("vhost-device-gpio-backend"),
-                backend.clone(),
-                GuestMemoryAtomic::new(GuestMemoryMmap::new()),
-            )
-            .map_err(Error::CouldNotCreateDaemon)?;
-
-            daemon.start(listener).unwrap();
-
-            match daemon.wait() {
-                Ok(()) => {
-                    info!("Stopping cleanly.");
+            match cfg {
+                GpioDeviceType::PhysicalDevice(device_num) => {
+                    let controller = PhysDevice::open(device_num).unwrap();
+                    start_device_backend(controller, socket.clone());
                 }
-                Err(vhost_user_backend::Error::HandleRequest(
-                    vhost_user::Error::PartialMessage | vhost_user::Error::Disconnected,
-                )) => {
-                    info!("vhost-user connection closed with partial message. If the VM is shutting down, this is expected behavior; otherwise, it might be a bug.");
+                #[cfg(feature = "mock_gpio")]
+                GpioDeviceType::SimulatedDevice(num_gpios) => {
```

Comment:
Shouldn't we add the `#[cfg]` part within this case of the `match` ? And so the below `#[cfg(not())]` can be written within, without the need of using `_ => {` case ?